File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.py
from math import sqrt
class Solution:
    def areaOfMaxDiagonal(self, dimensions: List[List[int]]) -> int:
        # dimensions = [[9,3],[8,6]]
        n=len(dimensions)
        diags=[]
        for i in range(n):
            l=dimensions[i][0]
            w=dimensions[i][1]
            diag=sqrt(l*l+w*w)
            diags.append([l,w,diag])

        maxlen = diags[0][2]
        maxarea = diags[0][0] * diags[0][1]
        for i in range(1, n):
            if diags[i][2] > maxlen or (diags[i][2] == maxlen and diags[i][0] * diags[i][1] > maxarea):
                maxlen = diags[i][2]
                maxarea = diags[i][0] * diags[i][1]
        return(maxarea)
                

File: 3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int maxarea=0; int maxdiag=0;
        for(auto &dim:dimensions){
            int l=dim[0]; int w=dim[1];
            int diag=pow(l,2)+pow(w,2);
            int area=l*w;
            if(maxdiag<diag){
                maxdiag=diag;
                maxarea=area;
            }else if(diag==maxdiag){
                maxarea=max(area,maxarea);
            }
        }
        return maxarea;
    }
};